librad/src/keys.rs

```diff
@@ -131,12 +123,13 @@ impl SecretKey {
                             Self::PKCS_ED25519_OID,
                         ));
                 });
-                let seed = yasna::construct_der(|writer| writer.write_bytes(&self.0[..32]));
+                let seed =
+                    yasna::construct_der(|writer| writer.write_bytes(&self.0.as_ref()[..32]));
                 writer.next().write_bytes(&seed);
                 writer
                     .next()
                     .write_tagged(yasna::Tag::context(1), |writer| {
-                        let public = &self.0[32..];
+                        let public = &self.0.as_ref()[32..];
```

Comment:
Does this not crash? The slice returned by `AsRef<[u8]>` is only 32 bytes. 

Iirc, the first thing written needs to be the secret part, and the second the public, which is commonly stored in one 64 bytes array. But not so with the zebra!

Reply:
Oh wow, it doesn't crash in the expected way, but the test was failing. Should be fixed in 650ba5f